xdsl/transforms/utils.py

```diff
@@ -0,0 +1,58 @@
+from xdsl.dialects import memref, scf
+from xdsl.ir import Block, Operation, SSAValue
+
+
+# TODO replace by functionality (when added) as described in https://github.com/xdslproject/xdsl/issues/2128
+def get_operation_at_index(block: Block, idx: int) -> Operation:
+    """Get an operation by its position in its parent block."""
+
+    for _idx, block_op in enumerate(block.ops):
+        if idx == _idx:
+            return block_op
+
+    raise ValueError(
+        f"Cannot get operation by out-of-bounds index {idx} in its parent block."
+    )
+
+
+def find_same_target_store(load: memref.Load):
```

Comment:
Possibly very dumb question, but there's no guarantee that the result of one memref.load operation is used by just one store op, is it? Shouldn't this function be returning a list of `found_ops`?

Reply:
Oh woops I see that this is in the comment of the PR, maybe also put these limitations in comments?

Reply:
That's why it returns `None` when there's more than one :)